pipenv/core.py

```diff
@@ -1574,6 +1548,166 @@ def pip_install(
     return c
 
 
+def pip_install_deps(
+    project,
+    deps=None,
+    allow_global=False,
+    no_deps=False,
+    pre=False,
+    dev=False,
+    selective_upgrade=False,
+    requirements_dir=None,
+    pypi_mirror=None,
+    trusted_hosts=None,
+    use_pep517=True,
+    use_constraint=False,
+):
+    if not trusted_hosts:
+        trusted_hosts = []
+    trusted_hosts.extend(os.environ.get("PIP_TRUSTED_HOSTS", []))
+    if not allow_global:
+        src_dir = os.getenv(
+            "PIP_SRC", os.getenv("PIP_SRC_DIR", project.virtualenv_src_location)
+        )
+    else:
+        src_dir = os.getenv("PIP_SRC", os.getenv("PIP_SRC_DIR"))
+    if not requirements_dir:
+        requirements_dir = vistir.path.create_tracked_tempdir(
+            prefix="pipenv", suffix="requirements"
+        )
+
+    hashed_requirements = tempfile.NamedTemporaryFile(
```

Comment:
From the pip documentation: https://pip.pypa.io/en/stable/topics/secure-installs/#hash-checking-mode
> Note that hash-checking is an all-or-nothing proposition. Specifying --hash against any requirement will activate this mode globally.
We have two types of requirements -- those normal once with hashes, and the editable, vcs, local types that do not and it is not possible to do this in a single call to pip, but we can do it in just two!